cmd/clair/shutdown.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"golang.org/x/sync/errgroup"
+)
+
+// Shutdown aggregates http.Sever Shutdown methods.
+type Shutdown struct {
+	mu sync.Mutex
+	m  map[string]*http.Server
+}
+
+// Add registers a server with specified name.
+//
+// Currently, it is disallowed to add two servers with the same name.
+func (s *Shutdown) Add(name string, srv *http.Server) {
```

Comment:
You could technically make the map key the server's pointer. You do not retrieve servers, and pointers will remain unique so naming conflicts go away.

Reply:
Oh, I did it that so that hot reloading could, eventually, work, but that can just quickly be reworked if it ever happens.